fvm/src/syscalls/bind.rs

```diff
@@ -114,9 +134,10 @@ macro_rules! impl_bind_syscalls {
                 if mem::size_of::<Ret::Value>() == 0 {
                     // If we're returning a zero-sized "value", we return no value therefore and expect no out pointer.
                     self.func_wrap(module, name, move |mut caller: Caller<'_, InvocationData<K>> $(, $t: $t)*| {
+                        charge_gas_for_exec_units(&mut caller)?;
                         let (mut memory, mut data) = memory_and_data(&mut caller)?;
                         let ctx = Context{kernel: &mut data.kernel, memory: &mut memory};
-                        Ok(match syscall(ctx $(, $t)*).into()? {
+                        let result = match syscall(ctx $(, $t)*).into()? {
```

Comment:
Note to self: If we trap here, we _don't_ need to "charge exec units for gas", because we won't be executing anything else.